File: sdk/python/src/ks_app/mcpproto/jsonrpc.py
"""JSON-RPC 2.0 工具函数和常量。

提供 MCP 协议层所需的 JSON-RPC 错误码和响应构造函数。
"""

from typing import Any

from starlette.responses import JSONResponse


# JSON-RPC 2.0 标准错误码
ERR_PARSE_ERROR = -32700
ERR_INVALID_REQUEST = -32600
ERR_METHOD_NOT_FOUND = -32601
ERR_INVALID_PARAMS = -32602
ERR_INTERNAL = -32603


def jsonrpc_result(req_id: Any, result: Any) -> JSONResponse:
    """构造 JSON-RPC 2.0 成功响应。"""
    return JSONResponse({"jsonrpc": "2.0", "id": req_id, "result": result})


def jsonrpc_error(req_id: Any, code: int, message: str) -> JSONResponse:
    """构造 JSON-RPC 2.0 错误响应。"""
    return JSONResponse({
        "jsonrpc": "2.0",
        "id": req_id,
        "error": {"code": code, "message": message},
    })

File: sdk/go/mcpproto/jsonrpc.go
package mcpproto

import "encoding/json"

// JSONRPCRequest JSON-RPC 2.0 请求
type JSONRPCRequest struct {
	JSONRPC string          `json:"jsonrpc"`
	ID      any             `json:"id,omitempty"`
	Method  string          `json:"method"`
	Params  json.RawMessage `json:"params,omitempty"`
}

// JSONRPCResponse JSON-RPC 2.0 成功响应
type JSONRPCResponse struct {
	JSONRPC string `json:"jsonrpc"`
	ID      any    `json:"id,omitempty"`
	Result  any    `json:"result"`
}

// JSONRPCError JSON-RPC 2.0 错误响应
type JSONRPCError struct {
	JSONRPC string          `json:"jsonrpc"`
	ID      any             `json:"id,omitempty"`
	Error   *JSONRPCErrBody `json:"error"`
}

// JSONRPCErrBody 错误详情
type JSONRPCErrBody struct {
	Code    int    `json:"code"`
	Message string `json:"message"`
}

const (
	errCodeParseError     = -32700
	errCodeInvalidRequest = -32600
	errCodeMethodNotFound = -32601
	errCodeInvalidParams  = -32602
	errCodeInternal       = -32603
)
